javalib/src/main/scala/java/util/ArrayDeque.scala

```diff
@@ -313,21 +359,25 @@ class ArrayDeque[E] private (initialCapacity: Int)
       status += 1
     empty = true
     startIndex = 0
-    endIndex = inner.length
+    endIndex = length()
   }
 
   private def firstIndexOf(o: Any): Int = {
     // scalastyle:off return
     if (isEmpty())
       return -1
-    val inner = this.inner // local copy
-    val capacity = inner.length // local copy
+    val innerJS = this.innerJS // local copy
+    val innerWasm = this.innerWasm // local copy
```

Comment:
Do I understand correctly that, these local copies are for better performance (accessing local copy is faster than class field), right?

Reply:
[note]
This is also a problem, let's wait until we decide to use `Array[AnyRef]` anyway.
If we don't, we need to `val innerJS = if (isWebAssembly) null else this.innerJS`, or maybe we can do the same as `PriorityQueue`.